### PR TITLE
Feature info based on promises.

### PR DESCRIPTION
**Pull request by mhaertwig**
Popup only for visible layers.

### CODE REVIEW

File: src/components/TopicLoader/TopicLoader.js [hunk 1]
```diff
@@ -57,7 +57,21 @@ class TopicLoader extends Component {
   }
 
   componentDidMount() {
+    const { dispatchSetClickedFeatureInfo, layerService, map } = this.props;
+
     this.updateLayers(this.topic.layers);
+
+    map.on('singleclick', e => {
```
Comment (oterral) [inline src/components/TopicLoader/TopicLoader.js]: Please remove the event on unmout with unByKey

File: src/components/TopicLoader/TopicLoader.js [hunk 2]
```diff
@@ -105,13 +105,6 @@ class TopicLoader extends Component {
       if (apiKey && flatLayers[i] instanceof TrafimageRasterLayer) {
         flatLayers[i].setApiKey(apiKey);
       }
-
-      if (
-        flatLayers[i] instanceof VectorLayer ||
-        flatLayers[i] instanceof WMSLayer
-      ) {
-        flatLayers[i].onClick(this.onClick.bind(this));
```
Comment (oterral) [inline src/components/TopicLoader/TopicLoader.js]: it seems we add  the same callbacks over and over without removing the old ones.

Comment (oterral) [reply src/components/TopicLoader/TopicLoader.js]: pff forget these lines are removed not added :D

**Comment by mhaertwig** [comment]
@oterral Thanks for the feedback. I addes some small commits and applied the requested changes.

**Pull request merged**